docs/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations_for_pandas.rst

```diff
@@ -3,17 +3,256 @@
+
+
+    To use the ``column_map_expectation`` decorator, your custom function must accept at least two arguments: ``self`` and ``column``. When the user invokes your Expectation, they will pass a string containing the column name. The decorator will then fetch the appropriate column and pass all of the non-null values to your function as a pandas ``Series``. Your function must then return a Series of boolean values in the same order, with the same index.
+
+    Custom functions can also accept additional arguments:
+
+    .. code-block:: python
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_values_to_be_less_than(self, column, value):
+            return column.map(lambda x: x<value)
+
+    Custom functions can have complex internal logic:
+
+    .. code-block:: python
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_value_word_counts_to_be_between(self, column, min_value=None, max_value=None):        
+            def count_words(string):
+                word_list = re.findall("(\S+)", string)
+                return len(word_list)
+
+            word_counts = column.map(lambda x: count_words(str(x)))
+
+            if min_value != None and max_value != None:
+                return word_counts.map(lambda x: min_value <= x <= max_value)
+            elif min_value != None and max_value == None:
+                return word_counts.map(lambda x: min_value <= x)
+            elif min_value == None and max_value != None:
+                return word_counts.map(lambda x: x <= max_value)
+            else:
+                return word_counts.map(lambda x: True)
+
+    Custom functions can reference external modules and methods:
+
+    .. code-block:: python
+
+        import pytz
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_values_to_be_valid_timezones(self, column, timezone_values=pytz.all_timezones):
+            return column.map(lambda x: x in timezone_values)
+
+    By convention, ``column_map_expectations`` always start with ``expect_column_values_...`` or ``expect_column_value_...`` (Ex: ``expect_column_value_word_counts_to_be_between``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+
+4. **Load some data**
+
+    To make your new Expectations available for validation, you can instantiate a ``MyCustomPandasDataset`` as follows:
+
+    .. code-block:: python
+
+        my_df = ge.read_csv("./data/Titanic.csv", dataset_class=MyCustomPandasDataset)
+
+    You can also coerce an existing pandas DataFrame to your class using ``from_pandas``:
+
+    .. code-block:: python
+
+        my_df = pd.read_csv("./data/Titanic.csv")
+        ge.from_pandas(my_other_df, dataset_class=MyCustomPandasDataset)
+
+    The same method can be used to coerce an existing Great Expectations DataAsset to your class, even though it's not actually from pandas:
```

Comment:
I think this means "pandas DataFrame" instead of "Great Expectations DataAsset"...but I'm not sure.

Reply:
No, I meant what I said. If you have a  `TypeXDataAsset` and you want to convert it to a `TypeYDataAsset`, the `from_pandas` method will do the trick.